python/TestHarness/TestHarness.py

```diff
@@ -698,7 +707,7 @@ def handleTestResult(self, tester, output, result, start=0, end=0, add_to_table=
     def cleanup(self):
         # Print the results table again if a bunch of output was spewed to the screen between
         # tests as they were running
-        if self.options.verbose or (self.num_failed != 0 and not self.options.quiet):
+        if (self.options.verbose or (self.num_failed != 0 and not self.options.quiet)) and not self.options.dry_run:
             print '\n\nFinal Test Results:\n' + ('-' * (TERM_COLS-1))
             for (test, output, result, timing, start, end) in sorted(self.test_table, key=lambda x: x[2], reverse=True):
                 print printResult(test, result, timing, start, end, self.options)
```

Comment:
when using dry run, we don't care about an re-organized re-printing of what didn't really happen yet...